ruby-challenges.rb

```diff
@@ -10,20 +10,93 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+# create a method named letter_contain that takes an array and a string as parameters
+def letter_contain(array, string)
+  # use .select to iterate over the array and remove values that do not meet certain conditions
+  # use variable word for the string being iterated over
+  # use .include? to see if the word has the string and print 
+  puts array.select {|word| word.include?(string)}
+end
+
+letter_contain(beverages_array, letter_o)
+# Output: 
+# coffee
+# soda water
+
+letter_contain(beverages_array, letter_t)
+# Output: 
+# water
+# soda water
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+def hash_value(hash)
+  puts hash.values.sort
+end
+
```

Comment:
Make sure to pay attention to the expected output, which is asking for an array of the values! Currently, it is outputting the values individually. Check out `.flatten` and see how you can use that in this challenge!